src/Sylius/Bundle/SettingsBundle/Manager/SettingsManager.php

```diff
@@ -71,29 +79,22 @@ class SettingsManager implements SettingsManagerInterface
      */
     protected $eventDispatcher;
 
-    /**
-     * @param SchemaRegistryInterface  $schemaRegistry
-     * @param ObjectManager            $parameterManager
-     * @param RepositoryInterface      $parameterRepository
-     * @param FactoryInterface         $parameterFactory
-     * @param Cache                    $cache
-     * @param ValidatorInterface       $validator
-     * @param EventDispatcherInterface $eventDispatcher
-     */
     public function __construct(
-        SchemaRegistryInterface $schemaRegistry,
-        ObjectManager $parameterManager,
-        RepositoryInterface $parameterRepository,
+        ServiceRegistryInterface $schemaRegistry,
+        ServiceRegistryInterface $resolverRegistry,
+        ObjectManager $settingsManager,
+        FactoryInterface $settingsFactory,
         FactoryInterface $parameterFactory,
-        Cache $cache,
+        SettingsResolverInterface $defaultResolver,
```

Comment:
The configuration option you've mentioned before is a good solution. Anyway, there comes another one:

 - let's say the default resolver is named `sylius.settings.default_resolver`, then the end user can "decorate" it without passing the "decorated" service by:

```xml
<service id="app.settings.default_resolver" class="Foo\Bar" decorates="sylius.settings.default_resolver">
  <argument type="service" id="custom.service" />
</service>
```

It's quite difficult to come up with a name for aliased services (like `basic` and `default` feels similar), but if we will name it meaningfully, that can be a nice solution too.